lib/storage.js

```diff
@@ -0,0 +1,35 @@
+'use strict';
+
+const storage = {};
+module.exports = exports = {};
+
+exports.createItem = function(schemaName, item){
+  if(!schemaName) return Promise.reject(new Error ('expected schema name'));
+  if (!item) return Promise.reject(new Error('expected item'));
+  if (!storage[schemaName]) storage[schemaName] = {};
+  storage[schemaName][item.id] = item;
+  return Promise.resolve(item);
+};
+
+exports.deleteItem = function(schemaName, dog){
+  if(!schemaName) return Promise.reject(new Error ('expected schema name'));
+  if (!dog) return Promise.reject(new Error('expected dog'));
+  if (!storage[schemaName]) storage[schemaName] = {};
+
+  storage[schemaName][dog.id] = '';
+
```

Comment:
Be aware that this isn't actually removing the dog from storage completely.  It is just setting the value of the property at that id to an empty string.  There is another way that completely 'Deletes' the property.